### PR TITLE
fix: Re-realize aspect files after plugin update

### PR DESCRIPTION
**Pull request by tpasternak**
Scenario
1. Import the project correctly
2. Download a new version of the plugin that contains new templates

Result:
The templates won't be re-realized, old java_info.bzl fil will persist

# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [ ] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: `<please reference the issue number or url here>`

# Description of this change

### CODE REVIEW

File: base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java [hunk 1]
```diff
@@ -37,76 +39,38 @@ public class SyncAspectTemplateProvider implements SyncListener {
...
-  private void copyFileTree(Path source, Path destination) throws IOException {
-    try (final var fileStream = Files.walk(source)) {
-      final var fileIterator = fileStream.iterator();
-      while (fileIterator.hasNext()) {
-        copyUsingRelativePath(source, fileIterator.next(), destination);
-      }
+    final var templateAspects = AspectRepositoryProvider.findAspectTemplateDirectory().orElse(null);
+    var javaTemplate = "java_info.template.bzl";
+    var realizedFile = realizedAspectsPath.resolve("java_info.bzl");
+    var templateWriter = new TemplateWriter(templateAspects.toPath());
```
[the remark this answers is not in the document]
Comment (LeFrosch) [reply base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java:65]: @tpasternak this still can case a NPE.

File: base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java [hunk 2]
```diff
@@ -37,76 +40,42 @@ public class SyncAspectTemplateProvider implements SyncListener {
...
-
-      var realizedFileName = templateFileName.replace(".template.bzl", ".bzl");
-      var realizedFile = aspectTemplateDir.resolve(realizedFileName);
-
-      if (!templateWriter.writeToFile(templateFileName, realizedFile, templateVariableMap)) {
-        throw new SyncFailedException("Could not create template for: " + language);
-      }
-    }
+  private static @NotNull Map<String, String> getStringStringMap(BlazeProjectDataManager manager) {
+    var projectData = Optional.ofNullable(manager.getBlazeProjectData()); // It can be empty on intial sync. Fall back to no lauguage support
```
Comment (LeFrosch) [inline base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java:73]: What does it mean for the initial sync that there is "no language support"?

File: base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeFrosch) [reply base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java:65]: This still can cause a NPE.

[the remark this answers is not in the document]
Comment (tpasternak) [reply base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java:66]: Ok, so throwing sync error

File: base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java (hunk 2 above)
Comment (LeFrosch) [reply base/src/com/google/idea/blaze/base/sync/aspects/strategy/SyncAspectTemplateProvider.java:74]: Makes sense, thanks for clarifying.

**Review by LeFrosch — APPROVED** [review]

**Comment by ujohnny** [comment]
so this is now merged, but probably we should have merged #6856 first though.

**Pull request merged**